Unit tests that rely on httpbin.org fail intermittently.

### Bug

The NetworkReader unit tests that use https://httpbin.org ([1](https://github.com/y-scope/clp/blob/1b419f51b983b8ac4199a41f505ebe680a68112a/components/core/tests/test-NetworkReader.cpp#L195), [2](https://github.com/y-scope/clp/blob/1b419f51b983b8ac4199a41f505ebe680a68112a/components/core/tests/test-NetworkReader.cpp#L235)) have been failing frequently ([example](https://github.com/y-scope/clp/actions/runs/14668146350/job/41168029430)), albeit intermittently due to httpbin.org being a bit unreliable. 

<details>
<summary>Unit test failure logs</summary>

```text
-------------------------------------------------------------------------------
network_reader_with_valid_http_header_kv_pairs
-------------------------------------------------------------------------------
/mnt/repo/components/core/tests/test-NetworkReader.cpp:195
...............................................................................

/mnt/repo/components/core/tests/test-NetworkReader.cpp:100: warning:
  Unexpected CURL error code: 22; expected: 0
  Error message:
  The requested URL returned error: 503

/mnt/repo/components/core/tests/test-NetworkReader.cpp:100: warning:
  Unexpected CURL error code: 22; expected: 0
  Error message:
  The requested URL returned error: 503

/mnt/repo/components/core/tests/test-NetworkReader.cpp:100: warning:
  Unexpected CURL error code: 22; expected: 0
  Error message:
  The requested URL returned error: 503

/mnt/repo/components/core/tests/test-NetworkReader.cpp:100: warning:
  Unexpected CURL error code: 22; expected: 0
  Error message:
  The requested URL returned error: 503

/mnt/repo/components/core/tests/test-NetworkReader.cpp:100: warning:
  Unexpected CURL error code: 22; expected: 0
  Error message:
  The requested URL returned error: 503

/mnt/repo/components/core/tests/test-NetworkReader.cpp:100: warning:
  Unexpected CURL error code: 22; expected: 0
  Error message:
  The requested URL returned error: 503

/mnt/repo/components/core/tests/test-NetworkReader.cpp:100: warning:
  Unexpected CURL error code: 22; expected: 0
  Error message:
  The requested URL returned error: 503

/mnt/repo/components/core/tests/test-NetworkReader.cpp:100: warning:
  Unexpected CURL error code: 22; expected: 0
  Error message:
  The requested URL returned error: 503

/mnt/repo/components/core/tests/test-NetworkReader.cpp:100: warning:
  Unexpected CURL error code: 22; expected: 0
  Error message:
  The requested URL returned error: 503

/mnt/repo/components/core/tests/test-NetworkReader.cpp:100: warning:
  Unexpected CURL error code: 22; expected: 0
  Error message:
  The requested URL returned error: 503

/mnt/repo/components/core/tests/test-NetworkReader.cpp:227: FAILED:
  REQUIRE( optional_content.has_value() )
with expansion:
  false
```
</details>

### CLP version

1b419f51b983b8ac4199a41f505ebe680a68112a

### Environment

Any GitHub runner.

### Reproduction steps

Run either of the [clp-core-build](https://github.com/y-scope/clp/blob/1b419f51b983b8ac4199a41f505ebe680a68112a/.github/workflows/clp-core-build.yaml) or [clp-core-build-macos](https://github.com/y-scope/clp/blob/1b419f51b983b8ac4199a41f505ebe680a68112a/.github/workflows/clp-core-build-macos.yaml) workflows.

--- a/components/core/tests/test-NetworkReader.cpp
+++ b/components/core/tests/test-NetworkReader.cpp
@@ -22,9 +22,11 @@
 #include "../src/clp/NetworkReader.hpp"
 #include "../src/clp/Platform.hpp"
 #include "../src/clp/ReaderInterface.hpp"
+#include "../src/clp/string_utils/string_utils.hpp"
 
 namespace {
 constexpr size_t cDefaultReaderBufferSize{1024};
+constexpr std::string_view cHttpHeadersEchoServerUrl{"https://mockhttp.org/headers"};
 
 [[nodiscard]] auto get_test_input_local_path() -> std::string;
 
@@ -192,10 +194,12 @@ TEST_CASE("network_reader_illegal_offset", "[NetworkReader]") {
     REQUIRE((clp::ErrorCode_Failure == reader.try_get_pos(pos)));
 }
 
+/**
+ * Sends some headers to an HTTP header echo server and validates that they're returned correctly in
+ * a JSON object under the "headers" key.
+ */
 TEST_CASE("network_reader_with_valid_http_header_kv_pairs", "[NetworkReader]") {
     std::unordered_map<std::string, std::string> valid_http_header_kv_pairs;
-    // We use httpbin (https://httpbin.org/) to test the user-specified headers. On success, it is
-    // supposed to respond all the user-specified headers as key-value pairs in JSON form.
     constexpr size_t cNumHttpHeaderKeyValuePairs{10};
     for (size_t i{0}; i < cNumHttpHeaderKeyValuePairs; ++i) {
         valid_http_header_kv_pairs.emplace(
@@ -204,12 +208,13 @@ TEST_CASE("network_reader_with_valid_http_header_kv_pairs", "[NetworkReader]") {
         );
     }
     std::optional<std::vector<char>> optional_content;
+
     // Retry the unit test a limited number of times to handle transient server-side HTTP errors.
     // This ensures the test is not marked as failed due to temporary issues beyond our control.
     constexpr size_t cNumMaxTrials{10};
     for (size_t i{0}; i < cNumMaxTrials; ++i) {
         clp::NetworkReader reader{
-                "https://httpbin.org/headers",
+                cHttpHeadersEchoServerUrl,
                 0,
                 false,
                 clp::CurlDownloadHandler::cDefaultOverallTimeout,
@@ -228,7 +233,11 @@ TEST_CASE("network_reader_with_valid_http_header_kv_pairs", "[NetworkReader]") {
     auto const parsed_content = nlohmann::json::parse(optional_content.value());
     auto const& headers{parsed_content.at("headers")};
     for (auto const& [key, value] : valid_http_header_kv_pairs) {
-        REQUIRE((value == headers.at(key).get<std::string_view>()));
+        // The echo server we're using returns the HTTP header names as lowercase (under the spec,
+        // they're case insensitive).
+        std::string key_lowercase{key};
+        clp::string_utils::to_lower(key_lowercase);
+        REQUIRE((value == headers.at(key_lowercase).get<std::string_view>()));
     }
 }
 
@@ -244,7 +253,7 @@ TEST_CASE("network_reader_with_illegal_http_header_kv_pairs", "[NetworkReader]")
             std::unordered_map<std::string, std::string>{{"Legal-Name", "CRLF\r\n"}}
     );
     clp::NetworkReader reader{
-            "https://httpbin.org/headers",
+            cHttpHeadersEchoServerUrl,
             0,
             false,
             clp::CurlDownloadHandler::cDefaultOverallTimeout,

